### ISSUE TITLE
JsonPatch: Cannot do an "add" patch to IList<Anything>

### ISSUE THREAD

**Issue by NTaylorMullen** [aspnet/Mvc#2280]
When attempting to patch `People` in the object below our patching mechanism throws.

```C#
public class Users
{
    public IList<string> People { get; } = new List<string>();
}
```

Note this happens for any type of `IList`

**Comment by rynowak** [comment]
@NTaylorMullen please include sample code of the controller/action and the jsonpatch document being sent to the server so we can verify.

@kirthik - there may be an issue with the detection for 'is this an array or a string'...

**Comment by NTaylorMullen** [comment]
```C#
public class HomeController : Controller
{
    public static Users Users = new Users();

    // GET: /<controller>/
    public IActionResult Index()
    {
        return View();
    }

    [HttpPatch]
    public string DoWork([FromBody] JsonPatchDocument<Users> patchDoc)
    {
        patchDoc.ApplyTo(Users);

        return JsonConvert.SerializeObject(Users);
    }
}

public class Users
{
    public IList<string> People { get; set; } = new List<string>();
}
```

```JavaScript
var data = [
    { "op": "add", "path": "People/1", "value": "Taylor"},
];

$.ajax({
    url: 'Home/DoWork',
    data: JSON.stringify(data),
    type: 'PATCH',
    dataType: 'json',
    contentType: 'application/json-patch+json'
});
```

**Comment by kirthik** [comment]
Fix checked in - 3bc7599903dfd8ddc55d9249794efa65b2f527e1
